library/core/src/char/methods.rs

```diff
@@ -159,15 +157,13 @@ impl char {
     /// `char`s. `from_u32_unchecked()` will ignore this, and blindly cast to
     /// `char`, possibly creating an invalid one.
     ///
-    /// [`u32`]: primitive.u32.html
-    ///
     /// # Safety
     ///
     /// This function is unsafe, as it may construct invalid `char` values.
     ///
     /// For a safe version of this function, see the [`from_u32`] function.
     ///
-    /// [`from_u32`]: #method.from_u32
+    /// [`from_u32`]: prim@char::from_u32()
```

Comment:
Oof, this is a bug in intra-doc links. `prim` should apply to the whole item, not to the type.
```
warning: incompatible link kind for `char::from_u32`
 --> tmp.rs:1:6
  |
1 | /// [char::from_u32()]
  |      ^^^^^^^^^^^^^^^^ help: to link to the builtin type, prefix with `prim@`: `prim@char::from_u32`
  = note: this link resolved to a builtin type, which is not a function
```

That said, I guess it works as-is, it's just inconsistent with the rest of intra-doc links ... @Manishearth what do you think? I think if we made it apply to the whole item there'd be no way to disambiguate between primitives and modules for associated items.

Reply:
Hmm, I think it could still use `core::primitive::char::from_u32` actually. That seems better anyway to remove the ambiguity.

I didn't know `core::primitive` existed when I added `prim@` ... maybe we should just get rid of that disambiguator altogether to be consistent with rustc_resolve?

Reply:
If you end up using `core::primitive` please ping me so I can update `intraconv` :)